commands/chain.go

```diff
@@ -108,3 +121,63 @@ var chainLsCmd = &cmds.Command{
 		}),
 	},
 }
+
+var chainBlockCmd = &cmds.Command{
+	Helptext: cmdkit.HelpText{
+		Tagline: "Show a filecoin block by its CID",
+		ShortDescription: `Prints the miner, parent weight, height,
+and nonce of a given block. If JSON encoding is specified with the --enc flag,
+all other block properties will be included as well.`,
+	},
+	Arguments: []cmdkit.Argument{
+		cmdkit.StringArg("cid", true, false, "CID of block to show"),
+	},
+	Options: []cmdkit.Option{
+		cmdkit.BoolOption("messages", "m", "show messages in block"),
+	},
+	Run: func(req *cmds.Request, re cmds.ResponseEmitter, env cmds.Environment) error {
+		cid, err := cid.Decode(req.Arguments[0])
+		if err != nil {
+			return err
+		}
+
+		block, err := GetPorcelainAPI(env).ChainGetBlock(req.Context, cid)
+		if err != nil {
+			return err
+		}
+
+		return re.Emit(block)
+	},
+	Type: types.Block{},
+	Encoders: cmds.EncoderMap{
+		cmds.Text: cmds.MakeTypedEncoder(func(req *cmds.Request, w io.Writer, block *types.Block) error {
```

Comment:
@anorth, @frrist  I had thought we were avoiding adding extra Text encoders.

Reply:
this is copy from old `show block` command. and almost all the command has `extra Text encoders`. so this text encoders should be one new issue to fix all commands.

Reply:
that's reasonable.